src/test/java/com/solvd/micro9/vkontaktah/service/PostServiceImplTest.java

```diff
@@ -0,0 +1,184 @@
+
+@ExtendWith(MockitoExtension.class)
+public class PostServiceImplTest {
+
+    @Mock
+    private PostRepository postRepository;
+
+    @InjectMocks
+    private PostServiceImpl postService;
+
+    @ParameterizedTest
+    @MethodSource("com.solvd.micro9.vkontaktah.service.TestDataProvider#getPosts")
+    void verifyAllPostsAreReceived(final List<Post> expectedPosts) {
+        Pageable pageable = PageRequest.of(0, 100);
+        Mockito.when(postRepository.findAll(Mockito.any(Pageable.class)))
+                .thenReturn(
+                        new PageImpl<>(
+                                expectedPosts,
+                                pageable,
+                                expectedPosts.size()
+                        )
+                );
+        List<Post> resultPosts = postService.getAll(pageable);
+        Assertions.assertNotNull(resultPosts);
+        Assertions.assertEquals(expectedPosts, resultPosts);
+    }
+
+    @ParameterizedTest
+    @MethodSource(
+            "com.solvd.micro9.vkontaktah.service.TestDataProvider#getPostsWithAuthorId"
+    )
+    void verifyPostsAreFoundByAuthor(
+            final List<Post> expectedPosts, final String authorId
+    ) {
+        Pageable pageable = PageRequest.of(0, 100);
+        Mockito.when(
+                        postRepository.findByAuthor(
+                                authorId, pageable.getOffset(), pageable.getPageSize()
+                        )
+                )
+                .thenReturn(expectedPosts);
+        List<Post> resultPosts = postService.findByAuthor(authorId, pageable);
+        Assertions.assertNotNull(resultPosts);
+        Assertions.assertEquals(expectedPosts, resultPosts);
+        resultPosts.forEach(
+                post -> Assertions.assertEquals(authorId, post.getAuthor().getId())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource(
+            "com.solvd.micro9.vkontaktah.service.TestDataProvider#getPostsWithLikerId"
+    )
+    void verifyPostsAreFoundByLiker(final List<Post> expectedPosts, final String likerId) {
+        Pageable pageable = PageRequest.of(0, 100);
+        Mockito.when(
+                        postRepository.findByLiker(
+                                likerId, pageable.getOffset(), pageable.getPageSize()
+                        )
+                )
+                .thenReturn(expectedPosts);
+        List<Post> resultPosts = postService.findByLiker(likerId, pageable);
+        Assertions.assertNotNull(resultPosts);
+        Assertions.assertEquals(expectedPosts, resultPosts);
+        resultPosts.forEach(post -> {
+            boolean isLikeSet = false;
+            for (Like like : post.getLikes()) {
+                if (likerId.equals(like.getUser().getId())) {
+                    isLikeSet = true;
+                    break;
+                }
+            }
+            Assertions.assertTrue(isLikeSet);
+        });
+    }
+
+    @ParameterizedTest
+    // CSOFF: LineLength
```

Comment:
@eyakauleva remove comments

Reply:
@h1alexbel these comments have functionality in themselves. They are put here to prevent these lines from being checked by checkstyle

Reply:
@h1alexbel I moved this suppression to the `checkstyle-supressions.xml`, but every time I modify this class and any lines added or removed, I have to edit suppression's line number in the file. So it seems comments were a better option. WDYT?

Reply:
done in 6ae03fe and then reverted in cb46629